src/utils/mixins.js

```diff
@@ -192,53 +245,24 @@ const fileUploader = {
       this.errorMessage = null;
       const hookResult = await preUpload();
       this.indeterminate = false;
-      const promiseList = this.files.map((file) => {
-        let promiseChain = Promise.resolve();
-        if (file.status === 'done') {
-          // We are resuming, skip already completed files
-          return promiseChain.then(() => file.result);
-        }
-        const progress = (event) => {
-          this.$set(file, 'progress', event);
-        };
-        file.status = 'uploading';
-        file.progress.indeterminate = true;
-        if (file.upload) {
-          promiseChain = promiseChain.then(() => file.upload.resume());
-        } else {
-          // eslint-disable-next-line new-cap
-          file.upload = new uploadCls(file.file, {
-            $rest: this.girderRest,
-            parent: (hookResult && hookResult.dest) ? hookResult.dest : dest,
-            progress,
-            params: file.uploadClsParams,
-          });
-          promiseChain = promiseChain
-            .then(() => file.upload.beforeUpload())
-            .then(() => file.upload.start());
+      const promiseList = [];
+      // shallow copy into queue
+      const fileQueue = this.files.slice().reverse();
+
+      /* Default to 5 workers */
+      const workerPool = [0, 1, 2, 3, 4];
+      await Promise.all(workerPool.map(async (workerId) => {
```

Comment:
Let's make this pool size a static constant.

```javascript
const WORKER_POOL_SIZE = 5;

const workerPool = [...Array(WORKER_POOL_SIZE)];
await Promise.all(workerPool.map(async () => {
```

Seems like we can get rid of `workerId` since it's not used.

Reply:
Yeah, that's better.